ClickHouse driver does not populate `md.Catalog` in `SourceMetadata`, causing `TestSQLDriver_CurrentSchemaCatalog/@sakila_ch25` to fail

## Summary

`go test ./...` on `master` (commit `96f801d3`) fails in `libsq/driver/driver_test.go:675` for the ClickHouse sakila source. The ClickHouse driver's `getSourceMetadata` sets `md.Schema` but never sets `md.Catalog`, so `metadata.Source.Catalog` is the zero value `""` when the test expects `"sakila"`.

All other SQL drivers (`sqlite3`, `postgres`, `sqlserver`, `mysql`) populate `md.Catalog` in their source-metadata flow; ClickHouse is the only driver missing this assignment.

## Reproduction

```bash
git rev-parse HEAD
# 96f801d397ca647785d5307f4878068266da8baf

go test ./libsq/driver/ -run TestSQLDriver_CurrentSchemaCatalog -v
```

Requires an accessible ClickHouse `@sakila_ch25` source (as configured in `testh/testdata/sources.sq.yml`, `clickhouse://sakila:***@localhost:9000?database=sakila`).

### Failure output (trimmed)

```text
--- FAIL: TestSQLDriver_CurrentSchemaCatalog (0.76s)
    --- FAIL: TestSQLDriver_CurrentSchemaCatalog/@sakila_ch25 (0.10s)
        driver_test.go:675:
            Error Trace: libsq/driver/driver_test.go:675
            Error:       Not equal:
                         expected: ""
                         actual  : "sakila"
            Test:        TestSQLDriver_CurrentSchemaCatalog/@sakila_ch25
FAIL    github.com/neilotoole/sq/libsq/driver   7.395s
FAIL
```

## Environment

- Commit: `96f801d3` (current `master`)
- Go: `go1.26.1 darwin/arm64`
- OS: macOS 15 (Darwin 24.6.0)
- ClickHouse source: `clickhouse://sakila:***@localhost:9000?database=sakila`

## Root Cause

`drivers/clickhouse/metadata.go` (lines 39-94), `getSourceMetadata` sets `md.Schema`, `md.Name`, `md.FQName`, but never assigns `md.Catalog`:

```go
md.DBVersion = version
md.DBProduct = "ClickHouse " + version
md.Schema = database
md.Name    = database
md.FQName  = database
md.User    = user
// md.Catalog is never set → "" (zero value)
```

For comparison, other drivers do populate it:

| Driver     | File                                   | Assignment                         |
|------------|----------------------------------------|------------------------------------|
| sqlite3    | `drivers/sqlite3/grip.go:113`          | `md.Catalog = "default"`           |
| postgres   | `drivers/postgres/metadata.go:210`     | `md.Catalog = md.Name`             |
| sqlserver  | `drivers/sqlserver/metadata.go:141`    | `md.Catalog = catalog`             |
| mysql      | `drivers/mysql/metadata.go:349,360`    | scanned from INFORMATION_SCHEMA    |
| clickhouse | `drivers/clickhouse/metadata.go`       | **missing**                        |

The ClickHouse README (`drivers/clickhouse/README.md:188`) already describes the intended mapping: "ClickHouse 'database' maps to SQ Schema/Catalog", and the test case at `libsq/driver/driver_test.go:660` encodes this:

```go
{sakila.CH, "sakila", "sakila"},  // {handle, wantSchema, wantCatalog}
```

So the expected behavior is that `md.Catalog == currentDatabase()` for ClickHouse, matching `md.Schema`.

## Failing Assertions

`libsq/driver/driver_test.go`:

- Line 669: `require.Equal(t, md.Schema, tc.wantSchema)` — passes for CH because `md.Schema = "sakila"`.
- Line 675: `require.Equal(t, md.Catalog, tc.wantCatalog)` — fails for CH because `md.Catalog = ""` but `tc.wantCatalog = "sakila"`.

Note: both assertions have the `require.Equal` argument order reversed (`require.Equal(t, expected, actual)` is the documented signature, but here the test passes `actual` first then `expected`). This isn't the cause of the bug, but it means the "expected" / "actual" labels in the failure output are swapped and can mislead triage. Worth fixing while here.

## Proposed Fix

In `drivers/clickhouse/metadata.go`, inside `getSourceMetadata`, set `md.Catalog = database` alongside the existing `md.Schema = database`:

```go
md.Schema  = database
md.Catalog = database
md.Name    = database
md.FQName  = database
```

Optional hygiene fix in `libsq/driver/driver_test.go` lines 669 and 675: swap the argument order to `require.Equal(t, tc.wantSchema, md.Schema)` / `require.Equal(t, tc.wantCatalog, md.Catalog)` so failure messages correctly label expected vs actual.

## Acceptance Criteria

1. `drivers/clickhouse/metadata.go` `getSourceMetadata` sets `md.Catalog` to the current ClickHouse database.
2. `go test ./libsq/driver/ -run TestSQLDriver_CurrentSchemaCatalog` passes for `@sakila_ch25`.
3. `go test ./...` passes (all existing ClickHouse tests remain green).
4. `make lint` is clean.
5. (Optional) `require.Equal` argument order corrected at `libsq/driver/driver_test.go:669` and `:675`.

## References

- `libsq/driver/driver_test.go:650-691` — `TestSQLDriver_CurrentSchemaCatalog`
- `drivers/clickhouse/metadata.go:39-94` — `getSourceMetadata`
- `drivers/clickhouse/clickhouse.go:503-523` — `CurrentCatalog` / `CurrentSchema`
- `drivers/clickhouse/README.md:188` — ClickHouse schema/catalog mapping
- Relates to #503 (ClickHouse driver initial implementation)
- Relates to #544 (ClickHouse type roundtrip)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,18 @@ Breaking changes are annotated with ☢️, and alpha/beta features with 🐥.
 > `v0.18.2`. This typically means that there was some CI/tooling mishap. Ignore
 > those gaps.
 
+## Unreleased
+
+### Fixed
+
+- [#555]: ClickHouse driver now populates `md.Catalog` in `SourceMetadata`,
+  matching the current ClickHouse database. Previously the field was left
+  empty, which caused `TestSQLDriver_CurrentSchemaCatalog` to fail for the
+  ClickHouse sakila source.
+
+[#555]: https://github.com/neilotoole/sq/issues/555
+
+
 ## [v0.50.0] - 2026-02-16
 
 ### Added

--- a/drivers/clickhouse/metadata.go
+++ b/drivers/clickhouse/metadata.go
@@ -56,6 +56,7 @@ func getSourceMetadata(ctx context.Context, src *source.Source, db sqlz.DB, noSc
 	md.DBVersion = version
 	md.DBProduct = "ClickHouse " + version
 	md.Schema = database
+	md.Catalog = database
 	md.Name = database
 	md.FQName = database
 	md.User = user

--- a/drivers/clickhouse/metadata.go
+++ b/drivers/clickhouse/metadata.go
@@ -32,7 +32,9 @@ import (
 //   - Driver/DBDriver: drivertype.ClickHouse
 //   - DBVersion: ClickHouse server version from version()
 //   - DBProduct: "ClickHouse" plus version string
-//   - Schema/Name/FQName: Current database from currentDatabase()
+//   - Schema/Catalog/Name/FQName: Current database from currentDatabase().
+//     ClickHouse uses a single "database" concept that maps to both schema
+//     and catalog in sq's model.
 //   - User: Current user from currentUser()
 //   - Size: Total database size from system.tables
 //   - Tables: Table metadata with TableCount/ViewCount (if noSchema is false)

--- a/libsq/driver/driver_test.go
+++ b/libsq/driver/driver_test.go
@@ -671,8 +671,8 @@ func TestSQLDriver_CurrentSchemaCatalog(t *testing.T) {
 			md, err := grip.SourceMetadata(th.Context, false)
 			require.NoError(t, err)
 			require.NotNil(t, md)
-			require.Equal(t, md.Schema, tc.wantSchema)
-			require.Equal(t, md.Catalog, tc.wantCatalog)
+			require.Equal(t, tc.wantSchema, md.Schema)
+			require.Equal(t, tc.wantCatalog, md.Catalog)
 
 			gotSchemas, err := drvr.ListSchemas(th.Context, db)
 			require.NoError(t, err)

